### DISCUSSION TITLE
Bookings not showing and confirmation email not sent

### DISCUSSION THREAD

**Discussion by JanineKall** [liedekef/events-made-easy#687] [Q&A]
I use EME on several sites but now something strange is happening. I was testing the email flow with Mollie test API. 
I made two bookings: one with payment transaction and one waiting to be paid. Emails 'waiting for approval' are sent, but confirmation email is not sent with payed booking.  Approval on payment is set. 
Event shows 1 approved booking and 1 booking to be approve. By trying to view the bookings, no data are found, not with the payed booking and not with the booking to be approved. 
Only thing I did differently with this one is that i set 'Create WP-user after booking'. Any ideas? https://rijkaantalent.nl

**Comment by liedekef** [comment]
I don't see EME active on that site, so not much to tell here :-)
What version of EME? PHP? Is there anything in the error logs etc ... ? What happens if you don't set "Create WP-user after booking" ?

**Reply by JanineKall** [reply]
Apologies, site was still in construction mode. I turned it off. Event link: https://rijkaantalent.nl/evenementen/rijk-aan-talent/. 
I tried again hoping construction mode caused the problem. I tried again without creating WP-user after booking. Still no change. PHP version is 8.1.31

**Comment by liedekef** [comment]
renaming the plugin folder is not recommended (should be events-made-easy). Next to that: what is the plugin version? And did you check the EME section "E-mail beheer", tab "Verzonden e-mails"? All EME mails sent go in there. Next: check also the EME mail queue settings and your EME mail settings in general.

**Reply by JanineKall** [reply]
It's weird that the folder has been renamed, I did not do that on purpose. Changing it back to normal removes EME from wordpress sidebar. 
All emails have been sent according to EME. Still only the 'To be approved' mails and the mails to contact person arrive. 
And... no data found when clicking on bookings. I am afraid I have to do it all over again? All those custom emails ... ;-(
Plugin version 2.5.20.  Downloaded it from github yesterday.

**Reply by liedekef** [reply]
No data when clicking on bookings ==> check the browser console for errors then, maybe something is causing a javascript error. If EME says the bookings are done, then that seems the only reason. But read on .... :-)
It seems to me you downloaded not the release zip from install but the main EME version which is my dev-version, sometimes that has bugs :-) If that is the case: update the version from github (it contains fixes). Next to that: best is to use a released version, but renaming the folder to just "events-made-easy" causes the plugin to be deactivated, so just reactivating it should solve that too.

**Reply by JanineKall** [reply]
I updated the plugin with the latest release. After that I checked the browser console. Indeed it shows errors: 
![console  errors](https://github.com/user-attachments/assets/752aa0a7-1172-4e90-bf04-79fa3e38dec4)

jQuery.Deferred exception: $_GET is not defined ReferenceError: $_GET is not defined
    at HTMLDocument.<anonymous> (https://rijkaantalent.nl/wp-content/plugins/events-made-easy/js/eme_admin_rsvp.js?ver=2.5.20:280:26)
    at e (https://rijkaantalent.nl/wp-admin/load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2:27028)
    at t (https://rijkaantalent.nl/wp-admin/load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2:27330) undefined
ce.Deferred.exceptionHook @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
t @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
setTimeout
(anoniem) @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
c @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
fireWith @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
fire @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
c @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
fireWith @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
ready @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2
P @ load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2Deze waarschuwing begrijpenAI
load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2 Uncaught ReferenceError: $_GET is not defined
    at HTMLDocument.<anonymous> (eme_admin_rsvp.js?ver=2.5.20:280:26)
    at e (load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2:27028)
    at t (load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,jquery-migrate,utils,jquery-ui-core,jquery-ui-mouse,jquery-ui-resizable,jquery-ui-draggable,jquery-ui-controlgroup,j&load%5Bchunk_1%5D=query-ui-checkboxradio,jquery-ui-button,jquery-ui-dialog,jquery-ui-sortable&ver=6.7.1:2:27330)

**Reply by liedekef** [reply]
That was an error in my development version, not in the public release. You now updated to the latest released version but your browser is still caching the javascript. Ctrl-refresh should do it.

**Reply by JanineKall** [reply]
yes yes yes!!!!!

**Reply by JanineKall** [reply]
but... still not sending approved booking email. Not even if i send it again to specific bookings.

**Reply by liedekef** [reply]
Is the option "Enable approved RSVP emails?" checked?

**Reply by JanineKall** [reply]
yep
![RSVP mails](https://github.com/user-attachments/assets/04ebe2ef-3086-4f10-af58-86cded802ab0)

**Reply by liedekef** [reply]
and are the templates for approved mails set? Both the subject and body  must not be empty

**Reply by liedekef** [reply]
if still not sure, feel free to send me a mail with temp admin account so I can take a look

**Reply by JanineKall** [reply]
I will do that, because according to EME all mails are sent. But the mails that are sent are the default emails, not my custom ones. But all mails triggered by approving booking do not arrive in mailboxes, while other mails do.
I'll sen you an admin account.

**Reply by JanineKall** [reply]
Should be in your inbox by now

**Reply by liedekef** [reply]
I can check for the content of the mails, but if EME says they are sent, then that means it is no longer in EME hands, but your mailserver. So you should check the mailserver logs for the reason as to why the mails are not sent.

**Reply by JanineKall** [reply]
Does that explain why only the approval mails do not arrive in inboxes?

**Reply by liedekef** [reply]
> Does that explain why only the approval mails do not arrive in inboxes?

no it does not, but if EME says the mails are sent, then it is not in EME's hands anymore but the mailserver. Maybe some spam filter or reason x or y ...

**Reply by liedekef** [reply]
several things:
- you're sending emails from ict@ihbv.nl (the owner of the event).
- the domain is rijkaantalent.nl
- the SPF dns records are not configured ok: the domain ihbv.nl is not specified to allow the IP rijkaantalent.nl to send emails on its behalf, so many servers will block this. Your mail logs will show this. 
- also it seems to me you need to take a look at the spf records for rijkaantalent.nl itself

I advise you to either fix your spf , set the owner or force a correct sender (in the EME settings)

**Reply by JanineKall** [reply]
Ok, work to do. Thanks for now.

**Comment by JanineKall** [comment]
Hi Franky, you are my hero once again.  Installed a SMTP mailer and now everything is working flawlessly. And again I learned to always insist on using my own favorite hosting party, instead of something else.

**Reply by liedekef** [reply]
EME can do SMTP too, no need for another plugin ... but anyway, glad it works.

**Comment by JanineKall** [comment]
Hi Franky, mail trouble again. Site has been working, bookings have been made and confirmed. But, suddenly, error message when sending EME mail. Log says mails have been sent. Test email gives back error and is not received. 
Your account is still valid. Do you know why this suddenly occurs?
![image](https://github.com/user-attachments/assets/b079262d-932a-4ac7-a280-2a1ed1aa19f7)

**Reply by liedekef** [reply]
I don't keep admin accounts :-) So I don't know the account info.
Anyway: this indicates a problem in your EME mail configuration, so check that.

**Reply by JanineKall** [reply]
Sorry to have bothered you. Never allow multiple admins in your website. They might change an email password without telling you. Solved it ;-(
